src/editor/contentState/tabCtrl.js

```diff
@@ -22,7 +22,70 @@ const tabCtrl = ContentState => {
     return false
   }
 
+  ContentState.prototype.isIndentableListItem = function () {
+    const { start, end } = this.cursor
+    const startBlock = this.getBlock(start.key)
+    if (startBlock.type !== 'p' || !startBlock.parent) {
+      return false
+    }
+
+    const listItem = this.getBlock(startBlock.parent)
+    if (listItem.type !== 'li' || start.key !== end.key || start.offset !== end.offset) {
+      return false
+    }
+
+    // Now we know it's a list item. Check whether we can indent the list item.
+    const list = this.getParent(listItem)
+    return list && /ol|ul/.test(list.type) && list.children.indexOf(listItem) >= 1
+  }
+
+  ContentState.prototype.indentListItem = function () {
+    const { start } = this.cursor
+    const startBlock = this.getBlock(start.key)
+    const listItem = this.getBlock(startBlock.parent)
+    const list = this.getBlock(listItem.parent)
+    const prevListItem = list.children[list.children.indexOf(listItem) - 1]
+
+    this.removeBlock(listItem)
+
+    // Search for a list in previous block
+    let newList = prevListItem.children.find(b => /ol|ul/.test(b.type))
```

Comment:
If the `newList` is the last child of its parent, we append listItem to its children.
else we should create a new list. eg.

```markdown
- hello world

  - nice to meet

  hello world

- nice to meet(Cursor)
```

After press tab:

```markdown
- hello world

  - nice to meet

  hello world

  - nice to meet
```